tests/Feature/PhotosAddHandlerTestAbstract.php

```diff
@@ -344,4 +345,37 @@ public function testTrickyVideoUpload(): void
 			],
 		]);
 	}
+
+	/**
+	 * Tests video upload without ffmpeg or exiftool.
+	 *
+	 * @return void
+	 */
+	public function testVideoUploadWithoutFFmpeg(): void
+	{
+		$hasExifTool = Configs::getValueAsInt(self::CONFIG_HAS_EXIF_TOOL);
+		Configs::set(self::CONFIG_HAS_EXIF_TOOL, 0);
+
+		$hasFFMpeg = Configs::getValueAsInt(TestCase::CONFIG_HAS_FFMPEG);
+		Configs::set(TestCase::CONFIG_HAS_FFMPEG, 0);
+
+		$response = $this->photos_tests->upload(
+			TestCase::createUploadedFile(TestCase::SAMPLE_FILE_GAMING_VIDEO)
+		);
+		$response->assertJson([
+			'album_id' => null,
+			'title' => 'gaming',
+			'type' => TestCase::MIME_TYPE_VID_MP4,
+			'size_variants' => [
+				'original' => [
+					'width' => 0,
+					'height' => 0,
+					'filesize' => 66781184,
+				],
+			],
```

Comment:
Maybe we should also assert that the other size variants are really `null`?


```suggestion
			'size_variants' => [
				'original' => [
					'width' => 0,
					'height' => 0,
					'filesize' => 66781184,
				],
				'medium2x' => null,
				'medium' => null,
				'small2x' => null,
				'small' => null,
				'thumb2x' => null,
				'thumb' => null,
			],
```

`FFMpeg` is always available to the the test suite. We only mimick a missing FFMpeg by disabling it in the configuration. If that fails (for some unknonwn reason) we might accidentally not test what we want, because FFmeg will be called without complain.

Reply:
Yeah, I agree that merely disabling it in the configuration is imperfect, but realistically that's probably all we can do in the testsuite?

Reply:
That's is why I considered checking the logs. This way, we can be sure that the exception has been triggered inside the code.